### ISSUE TITLE
local switches broken after being moved

### ISSUE THREAD

**Issue by RalfJung** [ocaml/opam#3672]
I moved around some directories, some of them containing local opam switches (`_opam` directories). Unfortunately, one of these switches are now broken: When entering the switch (which happens with this bash prompt hook after I `cd` there), opam adds the *old* location of the switch to the `$PATH`.

Curiously, the other switch being moved around worked fine. I just did `mv` on both of them so I do not know how opam managed to work in one case but break in another.

```
# opam config report
# opam-version      2.0.1 
# self-upgrade      no
# system            arch=x86_64 os=linux os-distribution=debian os-version=testing
# solver            builtin-mccs+glpk
# install-criteria  -removed,-count[version-lag,request],-count[version-lag,changed],-changed
# upgrade-criteria  -removed,-count[version-lag,solution],-new
# jobs              8
# repositories      4 (http), 4 (local), 1 (version-controlled) (default repo at 1ad6c60b)
# pinned            5 (rsync)
# current-switch    /home/r/Dokumente/Unisachen/iris/coq-users/gpfsl

```

**Comment by RalfJung** [comment]
Seems like the trouble is the `_opam/.opam-switch/environment` file, which did not get updated for one of these switches.

**Comment by RalfJung** [comment]
Hm, seems doing `opam update && opam upgrade` "fixes" this by recompiling everything.

Maybe the recompile is inevitable, but then opam should at least tell me that, instead of setting up a nonsense `$PATH`.

**Comment by rjbou** [comment]
Moving `_opam` directories is not handled by opam. better [export & import](https://opam.ocaml.org/doc/man/opam-switch.html#lbAE), or create a [link](https://opam.ocaml.org/doc/man/opam-switch.html#lbAE).
There were some problem regarding environment file not updated, it was corrected in 2.0.3 (cf #3691).

I close this issue, feel free to reopen it if there is something I needed.
Thanks!
